bucket/python.json

```diff
@@ -1,34 +1,40 @@
 {
     "homepage": "https://www.python.org/",
-    "license": "https://docs.python.org/3/license.html",
+    "description": "A programming language that lets you work quickly and integrate systems more effectively.",
+    "license": "Python-2.0",
     "version": "3.7.3",
+    "depends": "dark",
     "architecture": {
         "64bit": {
-            "url": "https://www.python.org/ftp/python/3.7.3/python-3.7.3-amd64.exe#/py3.exe",
+            "url": "https://www.python.org/ftp/python/3.7.3/python-3.7.3-amd64.exe",
             "hash": "md5:a2b79563476e9aa47f11899a53349383"
         },
         "32bit": {
-            "url": "https://www.python.org/ftp/python/3.7.3/python-3.7.3.exe#/py3.exe",
+            "url": "https://www.python.org/ftp/python/3.7.3/python-3.7.3.exe",
             "hash": "md5:ebf1644cdc1eeeebacc92afa949cfc01"
         }
     },
     "installer": {
-        "args": [
-            "/quiet",
-            "InstallAllUsers=$(@{$true=1;$false=0}[$global])",
-            "TargetDir=\"$dir\"",
-            "AssociateFiles=0",
-            "Shortcuts=0",
-            "InstallLauncherAllUsers=$(@{$true=1;$false=0}[$global])"
-        ],
-        "keep": true
+        "script": [
+            "&(file_path dark dark.exe) -nologo -x \"$dir\\_tmp\" \"$dir\\$fname\" | Out-Null",
+            "@('launcher.msi', 'path.msi', 'pip.msi') | ForEach-Object {",
+            "    Remove-Item \"$dir\\_tmp\\AttachedContainer\\$_\"",
+            "}",
+            "Get-ChildItem \"$dir\\_tmp\\AttachedContainer\\*.msi\" | ForEach-Object { extract_msi $_ $dir }",
```

Comment:
I am not sure if this will work, gci returns only file name since `toString()` will be called (most probably) when passing to function. 

So this will try to exract $_ inside current directory (where user is)

Reply:
i would go with:

```suggestion
            "(Get-ChildItem \"$dir\\_tmp\\AttachedContainer\\*.msi\").FullName | ForEach-Object { extract_msi \"$_\" \"$dir\" }",
```

Reply:
Or this should work too:

```suggestion
            "Get-ChildItem \"$dir\\_tmp\\AttachedContainer\\*.msi\" | ForEach-Object { extract_msi \"$($_.FullName)\" \"$dir\" }",
```

Reply:
Yeah to string will be called 
https://github.com/lukesampson/scoop/blob/131cf26b74c7aa74e2d8bbf39bd2f1205718a80a/lib/decompress.ps1#L21

So whole object will be converted into string. And it will look like this:
![image](https://user-images.githubusercontent.com/13260377/56658021-d1f47880-6699-11e9-8165-e661241a25d0.png)

Write-Output expect object and format it using table.
But write-host convert everything to string, which return only name

Reply:
I inserted `debug "$path"` in `extract_msi`, and found that `System.IO.FileSystemInfo` outputted by `Get-ChildItem` will transform to full path in double quote. So this is enough for extraction...

![python](https://user-images.githubusercontent.com/5832170/56673431-d72de380-66ea-11e9-881b-4896e2b6bd4b.png)

Reply:
What powershell version?

Reply:
And double quote is added inside `extract_` function, so it doesn't need to add them while calling function, `extract_msi $path $to` is okay here, white spaces will pass through.

Reply:
> 
> 
> What powershell version?

5.1 in Windows 10. I can't test in v3 now, but if https://github.com/lukesampson/scoop/pull/3330 is merged, there's no problem.

Reply:
That is not true. While powershell is executing `Invoke-Expression` (https://github.com/lukesampson/scoop/blob/master/lib/install.ps1#L1058), there are all variables "replaced" replaced with actual values.

So this:
`extract_7zip $dir`, which is valid in normal powershell
become this:
`extract_7zip C:\Users\UserName WIth Spaces\directory` which lead to error.

![Test](https://i.imgur.com/Jv6nyEB.png)

Reply:
That is the reason why every variable which contains path, needs to be quoted in all manifests

Reply:
PWSH 6.2.0 vs Powershell 5.1

![A](https://i.imgur.com/PVvMyxr.png)

Reply:
The function is `Invoke-Expression "lessmsi x "$path" "$to\""` (remove some \`, https://github.com/lukesampson/scoop/blob/master/lib/decompress.ps1#L52), and I check every extract function and every one quoted `$path` in double quote, like this:

![quote](https://user-images.githubusercontent.com/5832170/56675655-3f7ec400-66ef-11e9-884e-bcfe90d5950e.png)

Reply:
Yes, the full name of `$path` that passed through in function is strange, and it works in PS5 and PS6. I'm not sure why, may investigate tomorrow.

Reply:
Oh, I got your idea... I'll fix the quote😅

Reply:
You showed wrong line. 

Invoke expression will look like this:

```powershell
&(file_path dark dark.exe) -nologo -x \"C:\ User Space\\_tmp\" \"C:\ User Space\\$fname\" | Out-Null
@('launcher.msi', 'path.msi', 'pip.msi') | ForEach-Object {
    Remove-Item \"C:\User Sapce\\_tmp\\AttachedContainer\\$_\"
}
Get-ChildItem \"$dir\\_tmp\\AttachedContainer\\*.msi\" | ForEach-Object {
    extract_msi C:\User Space C:\User Space
}
```

Reply:
At least you conviced me to create new HyperV VM with username with spaces for testing :D